internal/inputs/clients/metrics.go

```diff
@@ -69,10 +69,52 @@ func NewMetricClient(ctx context.Context, projectId string, authToken string) (M
 		ctx:           ctx,
 		client:        client,
 		api:           api,
-		maxGoRoutines: defaultMaxGoroutines,
+		maxGoRoutines: maxGoroutines,
 	}, nil
 }
 
+type metricsClientBuilder struct {
+	ctx           context.Context
+	projectId     string
+	authToken     string
+	maxGoroutines int
+	timeout       int
+}
+
+func NewMetricsClientBuilder(ctx context.Context, projectId string, authToken string) *metricsClientBuilder {
+	return &metricsClientBuilder{
+		ctx:       ctx,
+		projectId: projectId,
+		authToken: authToken,
+	}
+}
+
+func (b *metricsClientBuilder) WithMaxGoroutines(maxGoroutines int) *metricsClientBuilder {
+	b.maxGoroutines = maxGoroutines
+	return b
+}
+
+func (b *metricsClientBuilder) WithTimeout(timeout int) *metricsClientBuilder {
+	b.timeout = timeout
+	return b
+}
+
+func (b *metricsClientBuilder) Build() (MetricsClient, error) {
+
+	var maxGoRoutines = defaultMaxGoroutines
+	if b.maxGoroutines != 0 {
+		maxGoRoutines = b.maxGoroutines
+	}
+
+	metricsClient, err := newMetricsClient(b.ctx, b.projectId, b.authToken, maxGoRoutines)
```

Comment:
We will need to find way to add our customer User Agent header to the Cloud Monitoring prometheus queries (for usage tracking purposes).
I couldn't find anything in prometheus API unfortunately.. The last resort solution seems to be crating custom `RoundTripper` adapter over the `AuthorizationCredentialsRoundTripper` and add header there on roundtrip.